File: js/9.js
// There exists exactly one Pythagorean triplet (a squared plus b squared equals c squared) for which _a_ + _b_ + _c_ = 1000.
// Find the product _abc_ <<<--->>>

function triplet(a, b, c) {
    if ((a*a + b*b) == c*c) {
        return true;
    } else {
        return false;
    }
}

function sums_to_100(a, b, c) {
    if (a + b + c == 1000) {
        return true;
    } else {
        return false;
    }
}

var res = -1;

for (var i = 1; i < 1000; i++) {
    for (var j = 1; j < 1000; j++) {
        for (var k = 1; k < 1000; k++) {
            if (sums_to_100(i,j,k)) {
                if (triplet(i,j,k)) {
                    console.log( i*j*k ); // 31875000
                }
            }
        }
    }
}

File: ruby/9.rb
# There exists exactly one Pythagorean triplet (a squared plus b squared equals c squared) for which _a_ + _b_ + _c_ = 1000.
# Find the product _abc_ <<<--->>>

1.upto(1000) { |a| 1.upto(1000) { |b| 1.upto(1000) { |c| 
  if a+b+c == 1000
    if a**2 + b**2 == c**2
      puts a * b * c
      return
    end
  end
}}}

# 31875000
# time ruby 9.rb
# 12.70s user 0.29s system 87% cpu 14.835 total